src/ir/datalog/value.h

```diff
@@ -37,7 +37,20 @@ class Value {
   virtual ~Value() {}
 };
 
-// Corresponds to Souffle's `number` type.
+// Corresponds to Souffle's `RamFloat` type.
+class Double : public Value {
```

Comment:
I'm not sure if this should be called `Double`, as it corresponds to `float` in Souffle. For all of the other types, we have preserved the name used in the Datalog language, only capitalizing it. It is pretty annoying that a type called `Float` would contain a `double`, but `double` does mean "double precision floating point value".